platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/AsyncTipsetEventCreationManager.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.tipset;
+
+import static com.swirlds.base.state.LifecyclePhase.NOT_STARTED;
+import static com.swirlds.base.state.LifecyclePhase.STARTED;
+import static com.swirlds.base.state.LifecyclePhase.STOPPED;
+
+import com.swirlds.base.state.Lifecycle;
+import com.swirlds.base.state.LifecyclePhase;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.threading.framework.BlockingQueueInserter;
+import com.swirlds.common.threading.framework.MultiQueueThread;
+import com.swirlds.common.threading.framework.config.MultiQueueThreadConfiguration;
+import com.swirlds.common.threading.futures.StandardFuture;
+import com.swirlds.common.threading.manager.ThreadManager;
+import com.swirlds.platform.internal.EventImpl;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+
+/**
+ * Manages the creation of events. Wraps a {@link SyncTipsetEventCreationManager} and provides an asynchronous
+ * interface.
+ */
+public class AsyncTipsetEventCreationManager implements Lifecycle {
```

Comment:
I split the business logic and threading apart to make the business logic simpler to unit test.